### PR TITLE
fix(me): exit non-zero when no token authenticates

### LINKED ISSUE #172: slck me exits 0 on auth failure (cli-common scriptability §3.1)
**What:** `slck me` returns nil (exit 0) when no tokens are valid.

**Where:** `internal/cmd/me/me.go:101-105` — prints "No valid tokens configured. Run 'slck init' to configure authentication." then `return nil`. The happy path (with valid tokens) works correctly.

**Why this is a divergence:** §3.1 of [`cli-common/docs/scriptability.md`](https://github.com/open-cli-collective/cli-common/blob/main/docs/scriptability.md) makes `me` the canonical scripted health check — it MUST exit non-zero when authentication is broken. A script doing `slck me || install_credentials` will silently skip the install because exit 0 == "I'm fine."

**Reference pattern:** nrq is the only CLI that gets this right today (`newrelic-cli/internal/cmd/initcmd/init.go` self-test path exits non-zero on auth failure).

**Scope:** Auth-failure path returns non-zero with a stderr error; success path unchanged. Update `me_test.go`. No CLI surface change.

Source: cli-common `docs/scriptability.md` §9, divergence catalog (PR open-cli-collective/cli-common#28).

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- When `runMe` finds both bot and user tokens missing/invalid, return an error instead of printing a message and `return nil`.
- The cobra `RunE` now propagates this to a non-zero exit, satisfying the cli-common scriptability §3.1 health-check contract.
- Happy path (at least one token authenticates) is unchanged.

## Test plan
- [x] Unit tests: `go test ./internal/cmd/me/...` — 7 tests green (2 flipped to assert error, 1 new boundary test at the cobra `Execute()` layer).
- [x] `TestRunMe_AuthFailed` gained `testutil.Setup(t)` so the nil userClient path no longer reaches the real keyring.
- [x] Lint: `make lint` — 0 issues.
- [x] Build: `make build` — `bin/slck` produced.
- [ ] Manual happy path: `slck me; echo \$?` → bot/user/workspace lines, exit 0.
- [ ] Manual failure path: clear tokens → `slck me; echo \$?` → stderr error \"no bot or user token authenticated; run 'slck init' to configure authentication\", exit non-zero.

## Breaking-change note
Any script that today assumed `slck me` always exits 0 will break. That script is *wrong* per the §3.1 contract — fixing it is the point of the change.

Closes #172

### CODE REVIEW

**Comment by rianjs** [comment]
## Review

No findings.

The diff stays tightly scoped to the product intent for slack-chat-api#172: `slck me` now fails non-zero when neither bot nor user authentication succeeds, while preserving the at-least-one-token-success behavior. The implementation avoids new abstraction, does not touch unrelated command surfaces, and the tests cover the key failure modes discussed in the plan:

- no configured tokens now returns an error
- invalid bot token plus hermetic empty user-token path now returns an error without touching the real keyring
- `NewCmd().Execute()` propagates the `RunE` error at the command boundary

Residual risk is low. The PR does not add a specific “one token invalid, the other valid” regression test, but the code path for partial success is unchanged and existing happy-path tests still cover bot-only and user-only success.

**Comment by rianjs** [comment]
Test coverage is adequate for the behavior change; one minor gap and one nit worth noting.

**Minor**
- `TestRunMe_AuthFailed` passes only `botClient` (non-nil, but auth fails) with `userClient=nil`, so the nil user path triggers `client.NewUserClient()` internally — which is correctly isolated by `testutil.Setup`. However, there's no symmetric test with `userClient` auth-failing and `botClient=nil`. The asymmetry is low-risk but leaves the user-token-only failure branch untested at the `runMe` level.

**Nit**
- `TestRunMe_BotTokenOnly` and `TestRunMe_UserTokenOnly` don't call `testutil.Setup`, so if `client.New()` / `client.NewUserClient()` were ever called on the nil path those tests could leak into the real keyring. Currently safe because non-nil clients are passed in, but worth hermetically isolating for consistency.
- The `assert.Contains` check on `"no bot or user token authenticated"` is appropriately scoped — not so tight it'll break on punctuation tweaks, not so loose it's meaningless.

**Happy path**
- `TestRunMe_BotTokenOnly`, `TestRunMe_UserTokenOnly`, and `TestRunMe_BothTokens` all assert `NoError`, confirming the at-least-one-success branch still returns nil. ✓

**Cobra contract**
- `TestNewCmd_NoTokens_ReturnsError` correctly exercises `cmd.Execute()` end-to-end with a hermetic keyring, proving the exit-code contract scripts depend on. ✓

File: internal/cmd/me/me_test.go [hunk 1]
```diff
@@ -53,11 +57,33 @@ func TestRunMe_UserTokenOnly(t *testing.T) {
...
+	opts := &meOptions{}
+
+	err := runMe(opts, nil, userClient)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "no bot or user token authenticated")
+}
+
 func TestRunMe_BothTokens(t *testing.T) {
 	botServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
```
Comment (monit-reviewer) [inline internal/cmd/me/me_test.go:89]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The PR adds TestRunMe_UserAuthFailedBotNil (user-auth-fails, bot=nil) to mirror TestRunMe_AuthFailed (bot-auth-fails, user=nil), but there is no test for the case where both clients are non-nil and both return invalid_auth. That is a distinct code path through the two auth calls and the nil-check, and a regression there would not be caught by any existing test.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `64d7110`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 1 |
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `internal/cmd/me/me_test.go:89`
> The PR adds TestRunMe_UserAuthFailedBotNil (user-auth-fails, bot=nil) to mirror TestRunMe_AuthFailed (bot-auth-fails, user=nil), but there is no test for the case where both clients are non-nil and both return invalid_auth. That is a distinct code path through the two auth calls and the nil-check, and a regression there would not be caught by any existing test.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**⚠️ Should Fix** - `internal/cmd/me/me_test.go:109`
> TestRunMe_NoTokens is missing testutil.Setup(t). This PR flipped its assertion from require.NoError to require.Error, but both clients are still passed as nil, so runMe calls client.New() and client.NewUserClient() against the real keyring. In any environment with valid credentials (developer machine or CI with injected tokens), those calls succeed, err is nil, and require.Error(t, err) fails. Every other nil-client test added or modified by this PR (BotTokenOnly, UserTokenOnly, AuthFailed, UserAuthFailedBotNil, NewCmd_NoTokens_ReturnsError) received testutil.Setup — this one was overlooked. Add testutil.Setup(t) as the first line of TestRunMe_NoTokens.

</details>

*3 info-level observations excluded. Run with `--verbose` to include.*

*2 PR discussion threads considered.*

---
<details>
<summary>Completed in 3m 26s | $0.94 | sonnet | daemon 0.2.121 | Glorfindel</summary>

| Field | Value |
|---|---|
| Model | sonnet |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude · sonnet |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 3m 26s wall · 9m 21s compute (Reviewers: 2m 45s · Synthesis: 39s) |
| Cost | $0.94 |
| Tokens | 239.7k in / 34.7k out |
| Turns | 13 |

**Per-workstream usage**

| Workstream | Model | In | Out | Cache read | Cache create | Cost |
|---|---|---|---|---|---|---|
| hybrid-synthesis | sonnet | 29.5k | 2.8k | 13.2k | 16.2k (1h) | $0.11 |
| harness-engineering:harness-architecture-reviewer | sonnet | 51.6k | 4.1k | 37.3k | 14.3k (1h) | $0.13 |
| harness-engineering:harness-enforcement-reviewer | sonnet | 37.8k | 7.8k | 15.1k | 22.7k (1h) | $0.21 |
| harness-engineering:harness-knowledge-reviewer | sonnet | 71.1k | 8.5k | 50.1k | 21.0k (1h) | $0.22 |
| harness-engineering:harness-self-documenting-code-reviewer | sonnet | 34.2k | 10.2k | 15.1k | 19.1k (1h) | $0.23 |
| security:security-code-auditor | haiku | 15.6k | 1.3k | 0 | 15.6k (1h) | $0.03 |

</details>

<sub><i>Re-reviews only run when `@monit-reviewer` is re-requested as a reviewer — push as many commits as you need, then re-request when ready. PRs targeting branches other than `main`, `master` are skipped, even when `@monit-reviewer` is re-requested.</i></sub>

*Note: Posted 1 inline comment. 1 finding remained in the summary because GitHub could not attach them to the current diff.*

File: internal/cmd/me/me_test.go (hunk 1 above)
Comment (rianjs) [reply internal/cmd/me/me_test.go:89]: Fixed — added TestRunMe_BothClientsAuthFailed covering the both-non-nil-both-invalid_auth path.

Comment (monit-reviewer) [reply internal/cmd/me/me_test.go:89]: **Settled**: monit-reviewer flagged a missing test case for when both clients are non-nil and both return invalid_auth. rianjs fixed it by adding TestRunMe_BothClientsAuthFailed covering that path.

File: internal/cmd/me/me_test.go [hunk 2]
```diff
@@ -100,10 +126,14 @@ func TestRunMe_NoTokens(t *testing.T) {
 
 	// Pass nil clients to trigger token lookup
 	err := runMe(opts, nil, nil)
-	require.NoError(t, err)
+	require.Error(t, err)
```
Comment (monit-reviewer) [inline internal/cmd/me/me_test.go:129]: 🟡 **Medium** (harness-engineering:harness-self-documenting-code-reviewer): TestRunMe_NoTokens is still missing testutil.Setup(t). This was flagged in the previous cycle (line 109) and remains unaddressed. The test passes nil for both clients, so runMe calls client.New() and client.NewUserClient() against the real OS keyring. With the assertion flipped to require.Error, any environment with valid credentials will see the real clients authenticate, err will be nil, and the assertion will fail. Every other nil-client test added or modified by this PR (TestRunMe_AuthFailed, TestRunMe_UserAuthFailedBotNil, TestRunMe_BothClientsAuthFailed, TestNewCmd_NoTokens_ReturnsError) correctly calls testutil.Setup(t). Add testutil.Setup(t) as the first line of TestRunMe_NoTokens.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `ad18968` | **Previous:** `64d7110` (incremental)

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**⚠️ Should Fix** - `internal/cmd/me/me_test.go:129`
> TestRunMe_NoTokens is still missing testutil.Setup(t). This was flagged in the previous cycle (line 109) and remains unaddressed. The test passes nil for both clients, so runMe calls client.New() and client.NewUserClient() against the real OS keyring. With the assertion flipped to require.Error, any environment with valid credentials will see the real clients authenticate, err will be nil, and the assertion will fail. Every other nil-client test added or modified by this PR (TestRunMe_AuthFailed, TestRunMe_UserAuthFailedBotNil, TestRunMe_BothClientsAuthFailed, TestNewCmd_NoTokens_ReturnsError) correctly calls testutil.Setup(t). Add testutil.Setup(t) as the first line of TestRunMe_NoTokens.

</details>

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 28s | $0.71 | sonnet | daemon 0.2.121 | Glorfindel</summary>

| Field | Value |
|---|---|
| Model | sonnet |
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude · sonnet |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 28s wall · 4m 02s compute (Reviewers: 1m 58s · Synthesis: 22s) |
| Cost | $0.71 |
| Tokens | 181.7k in / 15.7k out |
| Turns | 6 |

**Per-workstream usage**

| Workstream | Model | In | Out | Cache read | Cache create | Cost |
|---|---|---|---|---|---|---|
| hybrid-synthesis | sonnet | 35.6k | 1.3k | 13.2k | 22.3k (1h) | $0.11 |
| harness-engineering:harness-architecture-reviewer | sonnet | 22.7k | 2.2k | 4.0k | 18.6k (1h) | $0.10 |
| harness-engineering:harness-enforcement-reviewer | sonnet | 26.3k | 1.6k | 4.0k | 22.3k (1h) | $0.11 |
| harness-engineering:harness-knowledge-reviewer | sonnet | 27.1k | 1.0k | 4.0k | 23.1k (1h) | $0.10 |
| harness-engineering:harness-self-documenting-code-reviewer | sonnet | 28.8k | 8.4k | 4.0k | 24.7k (1h) | $0.22 |
| security:security-code-auditor | haiku | 20.5k | 1.1k | 0 | 20.4k (1h) | $0.03 |
| discussion-summarizer | — | 20.8k | 105 | 13.2k | 7.6k (1h) | $0.03 |

</details>

<sub><i>Re-reviews only run when `@monit-reviewer` is re-requested as a reviewer — push as many commits as you need, then re-request when ready. PRs targeting branches other than `main`, `master` are skipped, even when `@monit-reviewer` is re-requested.</i></sub>

File: internal/cmd/me/me_test.go (hunk 2 above)
Comment (rianjs) [reply internal/cmd/me/me_test.go:129]: False positive — TestRunMe_NoTokens (line 121) already calls testutil.Setup(t) on line 123, two lines above the line:129 anchor. The hermetic file backend has been in place since the test was first introduced; this PR only flipped the assertion from NoError to Error. Verified: `grep -n 'TestRunMe_NoTokens\|testutil.Setup' internal/cmd/me/me_test.go` shows line 121 (func) and line 123 (Setup call) adjacent.

**Comment by rianjs** [comment]
These are low-value, please approve the PR

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `ad18968`

### Human Override

Approved at the request of @rianjs ([comment](https://github.com/open-cli-collective/slack-chat-api/pull/174#issuecomment-4559657789)).

> These are low-value, please approve the PR

No automated reviewers were run.

**Comment by rianjs** [comment]
No architectural drift found.

The daemon edits are test-only and stay aligned with ticket 1’s product intent. Adding `testutil.Setup(t)` to the bot-only and user-only happy paths is actually an improvement: those tests also passed one client as `nil`, so they previously had the same latent “real keyring can leak in” problem. The added symmetric failure tests are scoped to `me` auth behavior and do not introduce new abstractions or unrelated command-surface changes.

Production behavior remains exactly the intended change: `slck me` returns an error only when neither bot nor user authentication succeeds; partial success still exits 0.
